plugins/home/src/homePageComponents/FeaturedDocs/FeaturedDocs.tsx

```diff
@@ -0,0 +1,173 @@
+
+import { makeStyles, Theme, Typography } from '@material-ui/core';
+
+/**
+ * Props customizing the <FeaturedDocs/> component.
+ *
+ * @public
+ */
+export type FeaturedDocsProps = {
+  /** The entity filter used to display only the intended item/s */
+  filter: EntityFilterQuery;
+  /** An optional color which can be customized through themes */
+  color?: 'inherit' | 'primary' | 'secondary';
+  /** An optional ClassNameMap created with makeStyles */
+  customStyles?: ClassNameMap<string>;
+  /** An optional ReactNode for empty states */
+  emptyState?: React.ReactNode;
+  /** An optional path to set for entity entry  */
+  path?: string;
+  /** An optional limit to set for link destination  */
+  responseLimit?: number;
+  /** An optional string to customize sublink text */
+  subLinkText?: string;
+  /** An optional string or ReactNode to customize the card title */
+  title?: React.ReactNode | string;
+};
+
+const useStyles = makeStyles<Theme>(
+  theme => ({
+    docDescription: {
+      fontSize: theme.typography.body1.fontSize,
+      fontWeight: theme.typography.body1.fontWeight,
+      marginBottom: theme.spacing(2),
+      marginTop: theme.spacing(2),
+    },
+    docSubLink: {
+      fontSize: theme.typography.subtitle1.fontSize,
+      fontWeight: theme.typography.subtitle1.fontWeight,
+      lineHeight: theme.typography.subtitle1.lineHeight,
+    },
+    docsTitleLink: {
+      fontSize: theme.typography.h6.fontSize,
+      fontWeight: theme.typography.h6.fontWeight,
+      lineHeight: theme.typography.h6.lineHeight,
+    },
+  }),
+  { name: 'BackstageFeaturedDocs' },
+);
+
+/**
+ * A component to display specific Featured Docs.
+ *
+ * @public
+ */
+export const FeaturedDocs = (props: FeaturedDocsProps) => {
+  const {
+    color,
+    customStyles,
+    emptyState,
+    filter,
+    path,
+    responseLimit,
+    subLinkText,
+    title,
+  } = props;
+  const linkText = subLinkText || 'LEARN MORE';
+  const defaultStyles = useStyles();
+  const styles = customStyles || defaultStyles;
+  const catalogApi: CatalogApi = useApi(catalogApiRef);
+  const {
+    value: entities,
+    loading,
+    error,
+  } = useAsync(async () => {
+    const response = await catalogApi.getEntities({
+      filter: filter,
+      limit: responseLimit || 10,
+    });
+    return response.items;
+  });
+
+  if (loading) {
+    return <Progress />;
+  }
+  if (error) {
+    return <ErrorPanel error={error} />;
+  }
+
+  return (
+    <InfoCard variant="gridItem" title={title || 'Featured Docs'}>
+      {entities?.length
+        ? entities.map(d => (
+            <div key={d.metadata.name} data-testid="docs-card-content">
+              <Link
+                className={styles.docsTitleLink}
+                data-testid="docs-card-title"
+                color={color}
+                to={
+                  path ||
+                  `/docs/${d.metadata.namespace || 'default'}/${d.kind}/${
+                    d.metadata.name
+                  }/`
```

Comment:
Hm it's unfortunate to form the link by hand like this, one should ideally use route refs because adopters aren't sure to have their docs page mounted on `/docs` at all.

Unfortunately the docRoot route ref is only exported from the techdocs plugin, not techdocs-react 🤔 https://github.com/backstage/backstage/blob/2cc1dfbaf58ea8a2db2700e60d7a4406ef4ce009/plugins/techdocs/api-report.md?plain=1#L316 that's actually unfortunate because plugins shouldn't import plugins.

OK for now, let's keep this as is, but we should work with @backstage/techdocs-maintainers to move that route ref out to become readable from techdocs-react so we can use it.

Reply:
understood, that is why I added the `path` prop to specify this if it is different.